packages/fast-components-react-base/src/stack-panel/stack-panel.props.ts

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import {
+    ManagedClasses,
+    StackPanelClassNameContract,
+} from "@microsoft/fast-components-class-name-contracts-base";
+
+export interface StackPanelManagedClasses
+    extends ManagedClasses<StackPanelClassNameContract> {}
+export interface StackPanelUnhandledProps extends React.HTMLAttributes<HTMLDivElement> {}
+
+export enum stackPanelOrientation {
+    horizontal = "horizontal",
+    vertical = "vertical",
+}
+
+export interface ItemSpanOverride {
+    [key: number]: number;
+}
+
+export interface StackPanelHandledProps extends StackPanelManagedClasses {
+    /**
+     * Whether items outside the viewport should be rendered or not
+     * default is true
+     */
+    enableVirtualization?: boolean;
+
+    /**
+     *  the indexes of children that should always be rendered
+     */
+    neverVirtualizeIndexes?: number[];
+
+    /**
+     *  The default span of each item (width in horizontal orientation, height in vertical)
+     */
+    defaultItemSpan?: number;
+
+    /**
+     * An object whose entries specifies the height of items that differ from the default height.
+     * Format is {itemIndex: heightValue}
+     */
+    itemSpanOverrides?: ItemSpanOverride;
+
+    /**
+     * How many items to preload before and after the range defined by viewportMaxHeight
+     */
+    preloadBufferLength?: number;
+
+    /**
+     * The orientation of the stackpanel
+     */
+    orientation?: stackPanelOrientation;
```

Comment:
Should we even have orientation? Can virtualization happen in both horizontal and vertical axis? Take excel for example.

Reply:
Yes, IMO we should have orientation.  A horizontal carousel could totally benefit from virtualization, for example,

Reply:
I'm in agreement, I meant what if it was virtualizing both horizontally and vertically.

Reply:
I was not intending to support virtualization on both axis simultaneously in this component.  The scenarios this is intended for initially is a feed and datagrid scenario which does not require more than one axis.